test/e2e/mcd_test.go

```diff
@@ -118,25 +120,39 @@ func TestMCDeployed(t *testing.T) {
 		return false, nil
 	})
 
-	listOptions := metav1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String(),
+	nodes, err := getNodesByRole(k, "worker")
+	if err != nil {
+		t.Error(err)
 	}
 
+	visited := make(map[string]bool)
 	err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
-		mcdList, err := k.CoreV1().Pods("openshift-machine-config-operator").List(listOptions)
-		if err != nil {
-			return false, err
-		}
-
-		for _, pod := range mcdList.Items {
-			res, err := k.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &v1.PodLogOptions{}).DoRaw()
+		for _, node := range nodes {
+			if visited[node.Name] {
+				continue
+			}
+			listOptions := metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String(),
+				FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String(),
+			}
+			mcdList, err := k.CoreV1().Pods("openshift-machine-config-operator").List(listOptions)
 			if err != nil {
-				// do not error out, we may be rebooting, that's why we list at every iteration
-				return false, nil
+				continue
 			}
-			for _, line := range strings.Split(string(res), "\n") {
-				if strings.Contains(line, "completed update for config "+newMCName) {
-					return true, nil
+			for _, pod := range mcdList.Items {
+				res, err := k.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &v1.PodLogOptions{}).DoRaw()
+				if err != nil {
+					// do not error out, we may be rebooting, that's why we list at every iteration
+					continue
+				}
+				for _, line := range strings.Split(string(res), "\n") {
+					if strings.Contains(line, "completed update for config "+newMCName) {
```

Comment:
Seems odd to parse the log files for this; why not verify the `currentConfig` annotation?

Reply:
Or for that matter, watch the worker machineconfigpool, which is doing just that.

Reply:
changed to watch for status changes for the pool (even if now it's intrinsically racy cause we may somehow be missing some status, for instance, we may very well miss the Updating status, or check the previous Updated status thinking all is good if we don't check Updating and so on).

Reply:
ok changed to check `currentConfig` instead.